### PR TITLE
Add v4 DDO structure validator

### LINKED ISSUE #583: DDO V4 schema validation

### PR DESCRIPTION
**Pull request by calina-c**
Closes #583. Works seamlessly with v3, can be merged to main.

### CODE REVIEW

**Comment by DMats** [comment]
@calina-c Can you add a list of the primary changes included in this PR to the PR description (the first comment)? I think that would make it easier to review.

File: aquarius/ddo_checker/schemas/v4/metadata_remote_v4.0.0.json [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+    "@context": {
+      "$id": "#/properties/@context",
+      "type": "array"
+    },
+    "id": {
+      "$id": "#/properties/id",
+      "type": "string",
+      "pattern": "^did\\:op\\:(.*)$"
+    },
+    "created": {
```
Comment (DMats) [inline aquarius/ddo_checker/schemas/v4/metadata_remote_v4.0.0.json:20]: `created` and `updated` are missing `type`

File: aquarius/ddo_checker/schemas/v4/metadata_remote_v4.0.0.json [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+                    "encryptionMode": {
+                      "$id": "#/properties/services/properties/items/files/properties/files/properties/encryptionMode",
+                      "type": "string",
+                      "default": "",
+                      "examples": ["CBC"]
+                    },
+                    "resourceId": {
+                      "$id": "#/properties/services/properties/items/files/properties/files/properties/resourceId",
+                      "type": "string",
+                      "default": 0,
```
Comment (DMats) [inline aquarius/ddo_checker/schemas/v4/metadata_remote_v4.0.0.json:324]: This default of `0` is not valid against the `"type": "string"`

File: aquarius/ddo_checker/schemas/v4/metadata_remote_v4.0.0.json [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+          "default": null
+        },
+        "algorithm": {
+          "$id": "#/properties/metadata/properties/algorithm",
+          "type": "object",
+          "required": ["container"],
+          "properties": {
+            "language": {
+              "$id": "#/properties/metadata/properties/algorithm/properties/language",
+              "type": "string"
```
Comment (DMats) [inline aquarius/ddo_checker/schemas/v4/metadata_remote_v4.0.0.json:177]: Most other keys of `"type": "string"` include a pattern that disallows multi-line strings `"pattern": "^(.*)$"`. This seems like a sensible restriction, so for consistency, I'd like to see it applied to all keys of `"type": "string"` in this file.

**Review by DMats — CHANGES_REQUESTED** [review]

**Comment by mihaisc** [comment]
`Status` is not updated https://github.com/oceanprotocol/aquarius/blob/792e59e53cf13859fad44a686ff3f66b6584fda8/aquarius/ddo_checker/schemas/v4/metadata_remote_v4.0.0.json#L437

https://deploy-preview-783--docs-oceanprotocol.netlify.app/concepts/did-ddo/#status

**Pull request merged**